### PR TITLE
Branching from image templates project

### PR DESCRIPTION
**Pull request by ChrisBr**
- Creating OBS:ImageTemplate attribute
- Branching from projects with this attribute will
  - set auto cleanup attribute
  - disable publish

### CODE REVIEW

File: src/api/app/models/branch_package.rb [hunk 1]
```diff
@@ -224,6 +224,10 @@ def create_branch_project
         tprj.relationships.build(user: User.current, role: Role.find_by_title!('maintainer'))
         tprj.flags.create(flag: 'build', status: 'disable') if @extend_names
         tprj.flags.create(flag: 'access', status: 'disable') if @noaccess
+        if params[:project]
+          project = Project.get_by_name(params[:project])
+          tprj.flags.create(flag: 'publish', status: 'disable') if project.try(:image_template?)
```
Comment (ChrisBr) [inline src/api/app/models/branch_package.rb]: Depending on the Configuration this is already done here:
https://github.com/openSUSE/open-build-service/blob/c1b60cf8675813a901a93e35d2720c7462a729fa/src/api/app/models/project.rb#L1278

Maybe it's is worth to move it there...

File: src/api/spec/models/branch_package_spec.rb [hunk 2]
```diff
@@ -31,5 +31,50 @@
...
+          branch_package.branch
+          project = Project.find_by_name(user.branch_project_name("openSUSE_Leap"))
+          expect(14.days.from_now - project.attribs.first.values.first.value.to_time).to be < 1.minute
+        end
+
+        it 'is set to the default' do
+          allow(Configuration).to receive(:cleanup_after_days).and_return(42)
+          branch_package.branch
+          project = Project.find_by_name(user.branch_project_name("openSUSE_Leap"))
+          expect(42.days.from_now - project.attribs.first.values.first.value.to_time).to be < 1.minute
```
Comment (ChrisBr) [inline src/api/spec/models/branch_package_spec.rb]: As discussed today, maybe a project.attribs.where better here.
Also maybe we should think about introducing TimeCop gem or something similiar...

File: src/api/spec/models/branch_package_spec.rb [hunk 3]
```diff
@@ -31,5 +31,50 @@
...
+      let(:leap_project) { create(:project, name: 'openSUSE_Leap') }
+      let(:apache) { create(:package, name: 'apache2', project: leap_project) }
+      let(:branch_package) { BranchPackage.new(project: leap_project.name, package: apache.name) }
+
+      context 'auto cleanup attribute' do
+        it 'is set to the default' do
+          allow(Configuration).to receive(:cleanup_after_days).and_return(42)
+          branch_package.branch
+          project = Project.find_by_name(user.branch_project_name("openSUSE_Leap"))
+          expect(42.days.from_now - project.attribs.first.values.first.value.to_time).to be < 1.minute
```
Comment (ChrisBr) [inline src/api/spec/models/branch_package_spec.rb]: As discussed today, maybe a project.attribs.where better here.
Also maybe we should think about introducing TimeCop gem or something similiar...

File: src/api/app/models/branch_package.rb (hunk 1 above)
Comment (hennevogel) [inline src/api/app/models/branch_package.rb]: absolutely

**Comment by hennevogel** [comment]
superseded by #2349

**Pull request closed without merging**